### PR TITLE
Vscode/vscodium add use wayland

### PR DESCRIPTION
**Pull request by TilCreator**
###### Motivation for this change
Wanting to use Vscode/Vscodium in Sway without Xwayland.

###### Things done
Add a `useWayland` option inspired by https://github.com/NixOS/nixpkgs/commit/bcad474c4ad57db4b21885973dc67b493f5730b1

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Synthetica9** [comment]
In my personal configuration I do 

```nix
      wrapProgram $out/bin/codium --add-flags \
        "--enable-features=UseOzonePlatform --ozone-platform=wayland"
```

which Is a bit nicer. This should also be used in `nixosTests.vscodium.wayland`

**Review by Synthetica9 — APPROVED** [review]
lgtm but I'd very much like a second opinion on this.

File: pkgs/applications/editors/vscode/vscodium.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, callPackage, fetchurl, nixosTests }:
+{ lib, stdenv, callPackage, fetchurl, nixosTests, useWayland ? false}:
```
Comment (turion) [inline pkgs/applications/editors/vscode/vscodium.nix:1]:
```suggestion
{ lib, stdenv, callPackage, fetchurl, nixosTests, useWayland ? false }:
```

**Comment by turion** [comment]
This is great, but maybe it should be documented somewhere so Wayland users will discover it?

**Comment by SuperSamus** [comment]
Doesn't this conflict with #147557?

**Comment by turion** [comment]
> Doesn't this conflict with #147557?

They both solve the same goal, and we probably only need one of the two. Which one it should be is probably a matter of discussion between Wayland users, @TilCreator @wmertens ?

**Comment by SuperSandro2000** [comment]
> Doesn't this conflict with #147557?

I am using wayland more by accident since plasma switched to it by accident and then kept wayland. I think it is very nice that I can just log into x11 and wayland with the same package set without a rebuild. Thats why I am in very strong favor of fixing this globally via the above quoted PR instead of on a per package basis.

**Comment by TilCreator** [comment]
I prefer #147557. Should I close the pr?

**Pull request closed without merging**